### PR TITLE
Rj tabbed interface

### PR DESCRIPTION
**Pull request by jrzricardo**

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    <h1>
+      <%= @user.username %>
+    </h1>
+
+    <ul class="nav nav-pills nav-justified">
+      <li class="nav-item">
+        <%= link_to "Posts", user_path(@user.username), class: "nav_link" %>
+      </li>
+      <li class="nav-item">
+        <%= link_to "Liked photos", liked_path(@user.username), class: "nav-link" %>
```
Comment (AdolfoNava) [inline app/views/users/show.html.erb]: good job handling the class here first try :)

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      </li>
+      <li class="nav-item">
+        <a class="nav-link" href="#">Feed</a>
+      </li>
+      <li class="nav-item">
+        <a class="nav-link" href="#">Followers</a>
```
Comment (AdolfoNava) [inline app/views/users/show.html.erb]: probably would get asked to turn these a tags to link_to

File: app/controllers/users_controller.rb
Comment (AdolfoNava) [inline app/controllers/users_controller.rb:1]: need the other methods for the 3 pages

File: config/routes.rb [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
   devise_for :users
   
   resources :comments
   resources :follow_requests
   resources :likes
   resources :photos
+
+  get ":username/liked" => "users#liked", as: :liked
+
+  get ":username" => "users#show", as: :user
```
Comment (AdolfoNava) [inline config/routes.rb:17]: Missing 3 other routes as mentioned above

**Review by AdolfoNava — CHANGES_REQUESTED** [review]
Refer to the notes

**Review by AdolfoNava — APPROVED** [review]
Looks good now

File: config/routes.rb [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
   resources :likes
   resources :photos
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get ":username/liked" => "users#liked", as: :liked
+  get ":username/feed" => "users#feed", as: :feed
+  get ":username/followers" => "users#followers", as: :followers
+  get ":username/following" => "users#following", as: :following
```
Comment (rezahusain) [inline config/routes.rb:15]: Really good naming consistency!

File: app/views/users/followers.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<%= render "users/navbar" %>
+
+<div class="row">
+  <div class="col-md-6 offset-md-3">
+    <div class="list-group">
+      <% @user.followers.each do |user| %>
+        <%= render "users/user", user: user %>
```
Comment (rezahusain) [inline app/views/users/followers.html.erb:7]: Love that your keeping your code shorter by utilizing partial templates.

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -65,6 +66,6 @@ def set_comment
 
     # Only allow a list of trusted parameters through.
     def comment_params
-      params.require(:comment).permit(:author_id, :photo_id, :body_text)
+      params.require(:comment).permit(:author_id, :photo_id, :body, )
```
Comment (rezahusain) [inline app/controllers/comments_controller.rb]: Don't know if the trailing comma at the end of body will break anything but just double check and make sure it doesn't.

**Review by rezahusain — APPROVED** [review]
Overall looks solid! Code is easy to read and concise.

**Review by CarlosCastanedaDev — APPROVED** [review]
Great job, Ricardo!

You can now merge these changes back to main.

**Comment by CodeX-y** [comment]
@photo = Photo.find(params[:id])

This is line 63 in your photos controller and it is causing an error. See if you can fix this error.

**Review by CodeX-y — CHANGES_REQUESTED** [review]
Everything looks good! All your bootstrap is in partials and routes look good.

**Review by CodeX-y — APPROVED** [review]
Everything looks good! All your bootstrap is in partials and routes look good.

**Review by CodeX-y — APPROVED** [review]

**Pull request merged**